### ISSUE TITLE
uncaught issue in restapi

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida_core#1390]
The current [schema](https://github.com/aiidateam/aiida_core/tree/develop/aiida/restapi/common) does not contain an entry for cifdata.

A request like `http://localhost:5000/api/v2/cifs/` therefore returns an internal server error.

The interesting thing is that there is a  test for the `cifs` endpoints and it passes:
https://github.com/aiidateam/aiida_core/blob/7b3113c7a9b998a8685e49e31a1cac8d2d2a997d/aiida/backends/tests/restapi.py#L792-L805

The reason seems to be that it is using the `app.test_client()`  (?).
@waychal We need to find a way of testing that the schema is complete.
Anyhow, it probably makes sense to wait until the schema has been integrated in the source code in a "proper" way.

**Comment by waychal** [comment]
In above test case schema is not required to download cif data. So if there is problem to get node schema, above test should and will pass.

We need to add separate testcases to check schema

**Comment by ltalirz** [comment]
@waychal Are you going to work on this soon?
I can easily fix the missing entry in the schema, but as I mentioned above I do not know how to make a general test that will detect missing entries in the schema.

**Comment by waychal** [comment]
@ltalirz 
url = self.get_url_prefix() + '/cifs/' + node_uuid + '/content/download'  
and http://localhost:5000/api/v2/cifs/7b3113c7a9b998a8685e49e31a1cac8d2d2a997d/content/download 
both are same. If test is working for this endpoint then it should also be working in browser unless file available for that node is not present or accessible and/or aiida orm is returning error.
With latest changes from develop branch, this url is working fine. Did you already fix the issue?

I guess we do not have any tests available for schema. Its better to add some.

**Comment by ltalirz** [comment]
@waychal 
> url = self.get_url_prefix() + '/cifs/' + node_uuid + '/content/download'  
> and http://localhost:5000/api/v2/cifs/7b3113c7a9b998a8685e49e31a1cac8d2d2a997d/content/download 
both are same.

It seems they aren't. 
Please upload a cif file to your DB and try.
If they are the same, how can the tests pass?

**Comment by waychal** [comment]
@ltalirz Error was in below if condition:
```python
-        if self.custom_schema is not None and 'columns' in self.custom_schema:
+        if self.custom_schema is not None and self.__label__ in self.custom_schema['columns'].keys():
             self._default_projections = self.custom_schema['columns'][
                 self.__label__]
         else:
             self._default_projections = ['**']
```
in case of testing, self.custom_schema is always None so its goes to else condition and test passes.

In browser, if you have custom_schema file in common folder, if condition becomes true (self.custom_schema is not None and 'columns' in self.custom_schema) and it gives internal server error for line:
self._default_projections = self.custom_schema['columns'][self.__label__]
If you do not have custom_schema file in common folder, everything works fine as in that case it uses else block.

I have fixed the if condition and created PR.
Let me know if you still have queries.

**Comment by ltalirz** [comment]
@waychal Thanks a lot for investigating this and finding the problem!
It's important we get these fixes out soon, so let's do this in `release_v0.12.1`

**Comment by sphuber** [comment]
Fixed in PR #1490
